lib/SIL/IR/SILFunctionType.cpp

```diff
@@ -4112,22 +4112,26 @@ TypeConverter::getLoweredFormalTypes(SILDeclRef constant,
       }
     }
 
-    // C++ operators that are implemented as non-static member functions get
-    // imported into Swift as static methods that have an additional
-    // parameter for the left-hand-side operand instead of the receiver object.
-    // These are inout parameters and don't get bridged.
-    // TODO: Undo this if we stop using inout.
-    if (auto method = dyn_cast_or_null<clang::CXXMethodDecl>(
-            constant.getDecl()->getClangDecl())) {
-      if (method->isOverloadedOperator()) {
-        bridgedParams.push_back(methodParams[0]);
-        methodParams = methodParams.drop_front(1);
+    auto partialFnPattern = bridgingFnPattern.getFunctionResultType();
+    for (unsigned i : indices(methodParams)) {
+      // C++ operators that are implemented as non-static member functions get
+      // imported into Swift as static methods that have an additional
+      // parameter for the left-hand-side operand instead of the receiver
+      // object. These are inout parameters and don't get bridged.
+      // TODO: Undo this if we stop using inout.
+      if (auto method = dyn_cast_or_null<clang::CXXMethodDecl>(
+              constant.getDecl()->getClangDecl())) {
+        if (i==0 && method->isOverloadedOperator()) {
+          bridgedParams.push_back(methodParams[0]);
+          continue;
+        }
       }
-    }
 
-    auto partialFnPattern = bridgingFnPattern.getFunctionResultType();
-    getBridgedParams(rep, partialFnPattern, methodParams, bridgedParams,
-                     bridging);
+      auto paramPattern = partialFnPattern.getFunctionParamType(i);
```

Comment:
@gribozavr Here is one more change that I made (I mean this whole additional commit). These few lines here are inlined from `getBridgedParams`, so that the index `i` is correct when calling `getFunctionParamType`.

Reply:
I see, makes sense.